q2cli/tests/test_usage.py

```diff
@@ -0,0 +1,257 @@
+            '    --m-metadata-file md1.tsv \\',
+            '    --m-metadata-file md2.tsv \\',
+            '    --o-out out.qza'
+        ),
+    ),
+    ###########################################################################
+
+    # identity with metadata column get metadata column example ###############
+    (
+        'identity_with_metadata_column',
+        'identity_with_metadata_column_get_mdc',
+        (
+            'qiime dummy-plugin identity-with-metadata-column \\',
+            '    --i-ints ints.qza \\',
+            '    --m-metadata-file md.tsv \\',
+            '    --m-metadata-column \'a\' \\',
+            '    --o-out out.qza'
+        )
+
+    ),
+    ###########################################################################
+
+    # optional inputs example #################################################
+    (
+        'optional_artifacts_method',
+        'optional_inputs',
+        (
+            'qiime dummy-plugin optional-artifacts-method \\',
+            '    --i-ints ints.qza \\',
+            '    --p-num1 1 \\',
+            '    --o-output output.qza',
+            'qiime dummy-plugin optional-artifacts-method \\',
+            '    --i-ints ints.qza \\',
+            '    --p-num1 1 \\',
+            '    --p-num2 2 \\',
+            '    --o-output output.qza',
+            'qiime dummy-plugin optional-artifacts-method \\',
+            '    --i-ints ints.qza \\',
+            '    --p-num1 1 \\',
+            '    --p-num2 None \\',
+            '    --o-output ints_b.qza',
+            'qiime dummy-plugin optional-artifacts-method \\',
+            '    --i-ints ints.qza \\',
+            '    --i-optional1 ints_b.qza \\',
+            '    --p-num1 3 \\',
+            '    --p-num2 4 \\',
+            '    --o-output output.qza'
+        )
+    ),
+    ###########################################################################
+
+    # variadic input example ##################################################
+    (
+        'variadic_input_method',
+        'variadic_input_simple',
+        ('qiime dummy-plugin variadic-input-method \\',
+         '    --i-ints ints_a.qza \\',
+         '    --i-ints ints_b.qza \\',
+         '    --i-int-set single_int1.qza \\',
+         '    --i-int-set single_int2.qza \\',
+         '    --p-nums 7 \\',
+         '    --p-nums 8 \\',
+         '    --p-nums 9 \\',
+         '    --o-output out.qza',
+         )
+    )
+]
+
+
+@pytest.mark.parametrize('action, example, exp', params)
+def test_render(dummy_plugin, action, example, exp):
+    action = dummy_plugin.actions[action]
+    use = CLIUsage()
+    action.examples[example](use)
+    cache = use.cache()
+    cache_renderer = CLIRenderer(cache)
+    rendered = '\n'.join(list(cache_renderer.render()))
+    assert rendered == '\n'.join(exp)
+
+
+class CliTests(unittest.TestCase):
+    def setUp(self):
+        get_dummy_plugin()
+        self.runner = CliRunner()
+
+    def test_examples(self):
```

Comment:
I just realized I put this test in the wrong place (I think) - if you agree, can you move over to `test_cli.py`? Once moved, this whole class (`CliTests`) can be removed.